File: src/main/java/sds/offre/infra/FormuleRepositoryEnPostgreSQL.java
package sds.offre.infra;

import sds.offre.concept_metier.Formule;
import sds.offre.concept_metier.FormuleRepository;
import sds.offre.concept_metier.IdFormule;
import sds.utils.concept_metier.Event;

import java.util.Collection;
import java.util.List;

public class FormuleRepositoryEnPostgreSQL implements FormuleRepository {


    @Override
    public Formule get(IdFormule id) {
        //SELECT * FROM ...
        throw new UnsupportedOperationException("Pas encore implémenté");
    }

    @Override
    public List<Event> addOrReplace(Formule formule) {
        //INSERT || UPDATE
        throw new UnsupportedOperationException("Pas encore implémenté");
    }

    @Override
    public Collection<Formule> getToutesLesFormules() {
        //SELECT * FROM ...
        throw new UnsupportedOperationException("Pas encore implémenté");
    }
}

File: src/main/java/sds/offre/infra/FormuleRepositoryEnPostgreSQL.kt
package sds.offre.infra

import sds.offre.concept_metier.Formule
import sds.offre.concept_metier.FormuleRepository
import sds.offre.concept_metier.IdFormule

class FormuleRepositoryEnPostgreSQL : FormuleRepository {


    override operator fun get(id: IdFormule): Formule {
        //SELECT * FROM ...
        throw UnsupportedOperationException("Pas encore implémenté")
    }

    override fun addOrReplace(formule: Formule) {
        //INSERT || UPDATE
        throw UnsupportedOperationException("Pas encore implémenté")
    }

    override fun toutesLesFormules(): Collection<Formule> {
        //SELECT * FROM ...
        throw UnsupportedOperationException("Pas encore implémenté")
    }
}
